### PR TITLE
feat(balancer) support for mTLS in health checks

### PR DESCRIPTION
**Pull request by locao**

### CODE REVIEW

File: kong/runloop/balancer.lua [hunk 1]
```diff
@@ -367,10 +367,18 @@ do
         checks.active.unhealthy.interval = 0
       end
 
+      local ssl_cert, ssl_key
+      if kong.configuration.client_ssl then
+        ssl_cert = kong.configuration.client_ssl_cert
```
Comment (dndx) [inline kong/runloop/balancer.lua:372]: We also support overriding client certificate used for upstream handshake on a per service basis. This should be supported in the health checker as well (possibly involving adding new field to the "Upstream" entity).

https://docs.konghq.com/2.0.x/admin-api/#add-service

File: kong/runloop/balancer.lua [hunk 2]
```diff
@@ -367,10 +367,31 @@ do
...
+      if upstream.client_certificate then
+        local get_certificate = require("kong.runloop.certificate").get_certificate
+        local cert, err = get_certificate(upstream.client_certificate)
+        if not cert then
+          log(ERR, "unable to fetch upstream client TLS certificate ",
+              upstream.client_certificate.id, ": ", err)
+          return nil, err
+        end
+        ssl_cert = cert.cert
+        ssl_key = cert.key
```
Comment (dndx) [inline kong/runloop/balancer.lua:380]: `cert.cert` and `cert.key` are `cdata` objects but the healthchecker expects strings. Will this work? I think the healthchecker should support `cdata` objects passed in directly as well.

File: kong/runloop/balancer.lua (hunk 1 above)
Comment (locao) [reply kong/runloop/balancer.lua]: Fixed!

File: kong/runloop/balancer.lua [hunk 3]
```diff
@@ -367,10 +367,31 @@ do
         checks.active.unhealthy.interval = 0
       end
 
+      local ssl_cert, ssl_key
+      if upstream.client_certificate then
+        local get_certificate = require("kong.runloop.certificate").get_certificate
```
[the remark this answers is not in the document]
Comment (locao) [reply kong/runloop/balancer.lua]: Fixed!

File: kong/runloop/balancer.lua (hunk 2 above)
Comment (locao) [reply kong/runloop/balancer.lua:380]: Nice catch! This was fixed in the healthchecker library, thanks!

**Comment by murillopaula** [comment]
can this PR be closed?

**Comment by murillopaula** [comment]
nvm, just saw that it's merged

**Pull request merged**